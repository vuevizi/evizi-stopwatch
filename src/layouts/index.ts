import MainLayout from "./MainLayout/MainLayout";
import Header from "./Header/Header";
import Footer from "./Footer/Footer";
export {Header,MainLayout,Footer}