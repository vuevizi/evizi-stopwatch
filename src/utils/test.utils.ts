export const findTestElement = (component: any, attr: string) => {
  const findElement: any = component.find(`[data-test="${attr}"]`);
  return findElement;
};
