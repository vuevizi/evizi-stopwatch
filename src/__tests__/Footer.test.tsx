import {shallow} from 'enzyme';
import {Footer} from '../layouts';
import {findTestElement} from '../utils/test.utils';
import toJson from 'enzyme-to-json';

const setUpComponent = (props = {}) => {
  const component: any = shallow(<Footer {...props} />);
  return component;
};

describe('Footer', () => {
  let component: any;
  beforeEach(() => {
    component = setUpComponent();
  });
  test('render with out crash', () => {
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
  test('render with right text', () => {
    const footerText = findTestElement(component, 'footer-text');
    expect(footerText.text()).toBe('Designed by Vu');
  });
});
