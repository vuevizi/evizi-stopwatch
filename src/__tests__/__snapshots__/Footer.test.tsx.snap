// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render with out crash 1`] = `
<styled.footer
  data-test="footer-container"
>
  <styled.h2
    data-test="footer-text"
  >
    Designed by Vu
  </styled.h2>
</styled.footer>
`;
